libvirt/utils_domain_def_test.go

```diff
@@ -42,13 +42,15 @@ func TestSplitKernelEmptyCmdLine(t *testing.T) {
 	}
 }
 
-func connect(t *testing.T) *libvirt.Connect {
-	conn, err := libvirt.NewConnect(os.Getenv("LIBVIRT_DEFAULT_URI"))
+func connect(t *testing.T) *libvirt.Libvirt {
+	conn := libvirt.Libvirt{}
+	URI := libvirt.ConnectURI(os.Getenv("LIBVIRT_DEFAULT_URI"))
+	err := conn.ConnectToURI(URI)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	return conn
+	return &conn
 }
 
 func TestGetHostArchitecture(t *testing.T) {
```

Comment:
Last failing test in `make testacc`:
```
--- FAIL: TestGetHostArchitecture (0.00s)
panic: assignment to entry in nil map [recovered]      
        panic: assignment to entry in nil map
```

Reply:
Fixed.

Reply:
mmmm, my fix works for `make testacc` but somehow I still get the problem when running a single test (`make testacc TEST_ARGS="-run SomeTest"`)

Reply:
When run `TestGetHostArchitecture` on its own? Interesting. I'll have a look shortly. 
Wonder if something in the debug logs.

Reply:
I only get it to happen when using `make testacc`:

```
$ make testacc TEST_ARGS="-test.v -test.run TestNetworkFromLibvirtError"
./travis/run-tests-acceptance -test.v -test.run TestNetworkFromLibvirtError
+ go test -v -covermode=count -coverprofile=profile.cov -timeout=1200s -test.v -test.run TestNetworkFromLibvirtError ./libvirt
=== RUN   TestNetworkFromLibvirtError
--- FAIL: TestNetworkFromLibvirtError (0.00s)
panic: interface conversion: interface {} is nil, not *libvirt.Client [recovered]
        panic: interface conversion: interface {} is nil, not *libvirt.Client

goroutine 18 [running]:
testing.tRunner.func1.2(0x12ae6e0, 0xc00047def0)
        /usr/lib64/go/1.16/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000582480)
        /usr/lib64/go/1.16/src/testing/testing.go:1146 +0x4b6
panic(0x12ae6e0, 0xc00047def0)
        /usr/lib64/go/1.16/src/runtime/panic.go:965 +0x1b9
github.com/dmacvicar/terraform-provider-libvirt/libvirt.TestNetworkFromLibvirtError(0xc000582480)
        /space/git/terraform-provider-libvirt/libvirt/network_def_test.go:134 +0x185
testing.tRunner(0xc000582480, 0x149b940)
        /usr/lib64/go/1.16/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
        /usr/lib64/go/1.16/src/testing/testing.go:1238 +0x2b3
FAIL    github.com/dmacvicar/terraform-provider-libvirt/libvirt 0.019s
FAIL
make: *** [Makefile:36: testacc] Error 1
```
(no matter if I define TF_ACC=1 LIBVIRT_DEFAULT_URI=qemu:///system  )

Now this works:

```
$ LIBVIRT_DEFAULT_URI=qemu:///system TF=ACC=1 go test -- -test.v -test.run TestGetHostCapabilties ./libvirt
PASS
ok      github.com/dmacvicar/terraform-provider-libvirt 0.034s
```

Reply:
Hmm, that single test run failure happens with other single test runs in that file for me too. 
I didn't check others, need to head out now. 

Great to get `TestGetHostCapabilities` passing. A `make testacc` went all the way through for me. :tada: